app/controllers/api_controller/providers.rb

```diff
@@ -5,9 +5,14 @@ module Providers
     CREDENTIALS_ATTR  = "credentials"
     AUTH_TYPE_ATTR    = "auth_type"
     DEFAULT_AUTH_TYPE = "default"
-    ENDPOINT_ATTRS    = %w(hostname ipaddress port)
+    ENDPOINT_ATTRS    = %w(hostname ipaddress port connection_configurations).freeze
     RESTRICTED_ATTRS  = [TYPE_ATTR, CREDENTIALS_ATTR, ZONE_ATTR, "zone_id"]
 
+    def show_providers
```

Comment:
It addes the `endpoints` attribute to `:additional_attributes` and then calls the default view method.

Return json without it looks like:
```
{
  "id": 122,
  "name": "Moki7",
  "type": "ManageIQ::Providers::Openshift::ContainerManager",
  "api_version": "v1",
  ....
}
```

Return json with it looks like:
```
{
  "id": 122,
  "name": "Moki7",
  "type": "ManageIQ::Providers::Openshift::ContainerManager",
  "api_version": "v1",
  "endpoints": [
    {
      "id": 290,
      "role": "default",
      "hostname": "ose-master.qa.lab.tlv.redhat.com",
      "port": 8443,
      "resource_type": "ExtManagementSystem",
      "resource_id": 122,
      "created_at": "2016-06-29T10:47:09Z",
      "updated_at": "2016-06-29T10:47:09Z",
      "verify_ssl": 1
    }, ... ],
   ...
}
```